lib/sles4sap/qesap/qesapdeployment.pm

```diff
@@ -1396,6 +1397,9 @@ sub qesap_cluster_logs {
             # Upload crm report
             qesap_upload_crm_report(host => $host, provider => $provider, failok => 1);
         }
+
+        # Collect logs in iscsi service node if there is.
+        qesap_save_y2logs(provider => $provider, host => 'iscsi[0]', failok => 1) if (script_run("grep -q 'iscsi' $inventory") == 0);
```

Comment:
Maybe the part `if (script_run("grep -q 'iscsi' $inventory") == 0);` is not needed. It is maybe fine to call Ansible with `iscsi[0]` also for test not having the iscsi VM, and let the filtering happening in Ansible itself.